conformance_tests/test_tiovx/test_ext_supplementary_data.c

```diff
@@ -0,0 +1,1435 @@
+    vx_image image = vxCreateImage(context, 10, 10, VX_DF_IMAGE_U8);
+    vx_rectangle_t rect = {.start_x = 0, .start_y = 0, .end_x = 9, .end_y = 9};
+    vx_image subimage = vxCreateImageFromROI(image, &rect);
+    ERROR_CHECK_VX_SUCCESS(vxSetSupplementaryUserDataObject((vx_reference)(image), exemplar), NULL);
+    vx_user_data_object childSupp = vxGetSupplementaryUserDataObject((vx_reference)(subimage), NULL, &status);
+    EXPECT_EQ_VX_STATUS(VX_SUCCESS, status);
+    vx_size size = sizeof(user_data_t);
+    vx_map_id map_id;
+    void *ptr;
+    ERROR_EXPECT_STATUS_BAD(vxSetUserDataObjectAttribute(childSupp, VX_USER_DATA_OBJECT_VALID_SIZE, &size, sizeof(size)), "Cannot set attribute of read-only supplementary data");
+    ERROR_EXPECT_STATUS_BAD(vxMapUserDataObject(childSupp, 0, size, &map_id, &ptr, VX_WRITE_ONLY, VX_MEMORY_TYPE_HOST, 0), "Cannot map read-only supplementary data for writing");
+    vx_image output1 = vxCreateImage(context, 10, 10, VX_DF_IMAGE_U8);
+    vx_image output2 = vxCreateImage(context, 10, 10, VX_DF_IMAGE_U8);
+    vx_graph graph = vxCreateGraph(context);
+    vx_image virt = vxCreateVirtualImage(graph, 10, 10, VX_DF_IMAGE_U8);
+    vx_kernel kernel = vxGetKernelByEnum(context, MY_USER_TEST_KERNEL);
+    EXPECT_VX_OBJECT(kernel, VX_TYPE_KERNEL);
+    vx_node node = vxCreateGenericNode(graph, kernel);
+    EXPECT_VX_OBJECT(node, VX_TYPE_NODE);
+    ERROR_CHECK_VX_SUCCESS(vxSetParameterByIndex(node, 0, (vx_reference)(image)), NULL);
+    ERROR_CHECK_VX_SUCCESS(vxSetParameterByIndex(node, 1, (vx_reference)(subimage)), NULL);
+    ERROR_CHECK_VX_SUCCESS(vxSetParameterByIndex(node, 2, (vx_reference)(output1)), NULL);
+    ERROR_CHECK_VX_SUCCESS(vxSetParameterByIndex(node, 3, (vx_reference)(virt)), NULL);
+    vx_node node2 = dumbCopyNode(graph, virt, output2);
+    ERROR_CHECK_VX_SUCCESS(vxProcessGraph(graph), NULL);
+    ERROR_EXPECT_STATUS(vxSetSupplementaryUserDataObject((vx_reference)(virt), exemplar), VX_ERROR_OPTIMIZED_AWAY, "Correctly failed to set supplementary data on virtual object outside graph after graph processing");
+    VX_CALL(vxReleaseKernel(&kernel));
+    VX_CALL(vxReleaseImage(&subimage));
+    VX_CALL(vxReleaseImage(&image));
+    VX_CALL(vxReleaseImage(&virt));
+    VX_CALL(vxReleaseImage(&output1));
+    VX_CALL(vxReleaseImage(&output2));
+    VX_CALL(vxReleaseUserDataObject(&exemplar));
+    VX_CALL(vxReleaseUserDataObject(&childSupp));
+    VX_CALL(vxReleaseNode(&node));
+    VX_CALL(vxReleaseNode(&node2));
+    VX_CALL(vxReleaseGraph(&graph));
+    tivxRemoveTargetKernelGeneral();
+}
+
+TESTCASE_TESTS(supplementary_data,
+                testRefCount,
+                testAllGetSet,
+                testInvalidTypes,
+                testGetSetErrors,
+                testChildren,
+                testCopySwapMove,
+                testExtend,
+                testExemplars,
+                testReadOnlyObjects, 
+                testObjectDescriptors)
```

Comment:
Please make sure this has a new line, it is not clear from the github view if it does